frame/asset-rate/src/lib.rs

```diff
@@ -0,0 +1,247 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! # Asset Rate Pallet
+//!
+//! - [`Config`]
+//! - [`Call`]
+//!
+//! ## Overview
+//!
+//! The AssetRate pallet provides means of setting conversion rates
+//! for some asset to native balance.
+//!
+//! The supported dispatchable functions are documented in the [`Call`] enum.
+//!
+//! ### Terminology
+//!
+//! * **Asset balance**: The balance type of an arbitrary asset. The network might only know about
+//!   the identifier of the asset and nothing more.
+//! * **Native balance**: The balance type of the network's native currency.
+//! * **Treasury spend**: A payment from the treasury after the corresponding proposal has been
+//!   approved.
+//!
+//! ### Goals
+//!
+//! The asset-rate system in Substrate is designed to make the following possible:
+//!
+//! * Whitelisting assets other than the native currency which can be accepted for Treasury spends.
```

Comment:
I think this shouldn't be documented here as a goal. I see pallet `asset-rate` as just a way to get a conversion from one asset to a default asset. That functionality is quite generic and could have a lot of uses outside of treasury spends.

Reply:
Also, if an asset rate is configured, that could serve as whitelisting for must purposes. Especially since we return an error if the asset does not exist.